packages/node_modules/nav-frontend-toggle/src/toggle-gruppe.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import * as PT from 'prop-types';
+import * as cn from 'classnames';
+import { omit, guid } from 'nav-frontend-js-utils';
+
+import ToggleKnappPure, { ToggleKnappPureProps, ToggleKnappPurePropsShape } from './toggle-knapp-pure';
+
+import 'nav-frontend-toggle-style';
+
+export interface ToggleGruppeProps {
+    /**
+     * Array av toggle knapper, se `toggle-knapp-pure.tsx`
+     */
+    toggles?: ToggleKnappPureProps[];
+    /**
+     * Egendefinert klassenavn.
+     */
+    className?: string;
+    /**
+     * Hvis `true` vil det være mulig å velge flere knapper om gangen.
+     */
+    multi?: boolean;
+    /**
+     * Angir hvilke knapper som skal være trykket inn som default.
+     */
+    defaultPressed?: number | number[];
+    /**
+     * Hvis 'true' reduseres høyre/venstre-padding på knappene betraktelig.
+     */
+    kompakt?: boolean;
+    /**
+     * Custom onChange handler
+     */
+    onChange?: (event: React.SyntheticEvent<EventTarget>, indexes: number[]) => void;
+}
+
+export interface ToggleGruppeState {
+    pressedIndexes: number[];
```

Comment:
Hva skjer med denne hvis innholdet i `toggles` eller `children`-propsene endrer seg? Dette ville jeg nok også styrt med å se på `pressed`-verdiene på propsene til `ToggleKnapp`-instansene, istedetfor å styre med state for hva som er valgt på `ToggleGruppe`. Dette er det jo `ToggleKnapp`-ene som skal forholde seg til, `ToggleGruppe` trenger bare å sende verdiene videre opp i DOM :)

Reply:
Denne er fremdeles aktuell @Lillebo. Hva skjer når pressed-verdiene i `toggles` endrer seg? Er redd denne typen styring av state kan gå ganske galt, fordi man styrer samme state på to steder: 1) med pressed-props som brukes på toggleinstanser, og 2) med pressedIndexes, som er en liste med indekser som ligger på state i `ToggleGruppe`. Hver endring i state på det ene stedet, må føre til en tilsvarende endring på det andre, hvis ikke vil det gå galt, så det blir ganske skjørt. Er det noe her som gjør at det ikke holder å styre pressed-verdiene på `ToggleKnapp`?

Reply:
En annen ting er som er relatert til det samme er om vi egentlig trenger pure/ikke-pure-varianter, for state'n på skjemaelementer generelt styres vel utelukkende i komponentene som bruker dem, ikke internt i skjemaelementene. Tanker @Lillebo?

Reply:
Hm, merk forskjellen på pure og ikke-pure versjon av komponenten. 

`ToggleGruppe` (ikke-pure) har intern state for at brukeren skal få slippe å implementere reglene for `multiSelect` osv. manuelt, og trenger kun å forholde seg til `onChange`-callbacket. I denne komponenten vil `toggles` propen kun være av interesse ved mount for å lese hvilke knapper som skal være `pressed` som default. Eventuelle endringer i `toggles`-propen etter mount vil bli ignorert ettersom komponenten håndterer dette internt og rapporterer endringene ut via `onChange`.

For `ToggleGruppePure` vil det fungere slik du beskriver; at `toggles`-arrayet blir oppdatert av en eller annen ekstern state manager (som en reaksjon på klikk) og formidlet tilbake med nye verdier til komponenten som da re-rendrer i henhold.

Reply:
Hva mener du med at det blir ignorert? Endringer i `toggles`-propen på `ToggleGruppe` vil vel bli like mye re-rendret som endringer i `toggles`-propen på `ToggleGruppePure`, med mindre det er noe jeg ikke ser?

Reply:
Ikke når det kommer til `pressed` verdiene. Internt i komponenten blir de kun hensyntatt ved mount ([ref](https://github.com/navikt/nav-frontend-moduler/pull/378/files#diff-abb61d82432721d3f94c364d29af554fR42)). Deretter får knappene sine pressed-props bare fra intern state ([ref](https://github.com/navikt/nav-frontend-moduler/pull/378/files#diff-abb61d82432721d3f94c364d29af554fR84)).

Reply:
Nei, ikke hvis du endrer `pressed`-verdiene alene, siden `toggles` i seg selv da fortsatt vil være den samme selv om verdien langt ned i objektstrukturen har endret seg.  

Men om du faktisk endrer på `toggles` (f.eks. at parent-komponenten lager en helt ny liste med andre elementer og sender den som `toggles`-prop), vil hele komponenten re-rendres, og det som state'n i komponenten tar utgangspunkt i, i kodeeksemplet her:

```
        const toggles = this.props.toggles.map((toggle, i) => ({
            ...toggle,
            kompakt: this.props.kompakt,
            pressed: this.state.pressedIndexes.indexOf(i) !== -1,
            onClick: (e) => this.handleClick(e, i, toggle.onClick)
        }));
```

Vil være utdatert, siden den baserer seg på forrige sett av props. Så state'n her må vel på en eller annen måte oppdateres?

Reply:
Svar på Edit @Lillebo: Om ☝️ er tenkt å ikke ha noe effekt på komponenten, bør vi endre navnet på property-en til noe annet enn `toggles`, tenker det er en forventning om at endring på props fører til endring av resultat og ny render, utenom i helt spesielle tilfeller som f.eks. defaultValue. `defaultToggles`?

Reply:
Hm, gode spørsmål. Må tenke litt 🙂

Reply:
Sliter litt med å skjønne hvilket scenario der `toggles`-propen behøver å endres etter at en instans av `ToggleGruppe` er mounted. Endringer av views f.eks. vil jo føre til re-mounts, hvor denne problematikken ikke vil gjelde.

Uansett, lurer på om det beste hadde vært å bare re-introdusere `defaultPressed` propen på `ToggleGruppe` i stedet for å utlede dette fra `toggles`-propen. Det vil i så fall gjøre det relativt trivielt å nullstille/oppdatere intern state for å samsvare med eventuelle endringer på `toggles` ved å sjekke om `defaultPressed` har endret seg i `componentWillReceiveProps`.

Reply:
Litt usikker på om jeg skjønner hva som menes her @Lillebo. Hvis jeg forstår dette rett, er vel den initielle verdien til `toggles` på sett og vis det samme som `defaultPressed`, siden objektene i `toggles` allerede har property-en `pressed`.

Ref denne kommentaren: 
https://github.com/navikt/nav-frontend-moduler/pull/378#discussion_r224417919

Tanken bak min forrige kommentar var at hvis det er meningen at endringer i `toggles`-propen ikke skal føre til nye renders av `ToggleGruppe`, fordi alt da vil styres av intern state, så kan vel denne prop-en like gjerne hete `defaultToggles`. Da blir det mer tydelig at endring av denne propen (utenom ved mount) ikke vil endre på noe.